src/frontend/src/helpers/Storage/LocalStorage.ts

```diff
@@ -56,9 +59,10 @@ class LocalStorage implements AbstractStorage {
     * @param {string} file_history: The history of the file.
     * @param {string || any false value} checksum : The online checksum of the file, false to not update (offline write).
     */
-  public async writeFile(fid: FileID, contents?: string): Promise<void> {
```

Comment:
content is undefined when a file is unavailable offline -- note that writeFile has to deal with this case as syncAll will be called with binary files.